gobblin-cluster/src/test/java/org/apache/gobblin/cluster/HelixAssignedParticipantCheckTest.java

```diff
@@ -0,0 +1,123 @@
+import org.apache.gobblin.configuration.ConfigurationKeys;
+import org.apache.gobblin.testing.AssertWithBackoff;
+
+
+public class HelixAssignedParticipantCheckTest {
+  private IntegrationJobSuite suite;
+  private HelixManager helixManager;
+  private Config helixConfig;
+
+  @BeforeClass
+  public void setUp()
+      throws Exception {
+    //Set up a Gobblin Helix cluster
+    suite = new IntegrationJobSuite();
+
+    helixConfig = suite.getManagerConfig();
+    String clusterName = helixConfig.getString(GobblinClusterConfigurationKeys.HELIX_CLUSTER_NAME_KEY);
+    String zkConnectString = helixConfig.getString(GobblinClusterConfigurationKeys.ZK_CONNECTION_STRING_KEY);
+    helixManager = HelixManagerFactory.getZKHelixManager(clusterName, "TestManager",
+        InstanceType.SPECTATOR, zkConnectString);
+
+    //Connect to the previously started Helix cluster
+    helixManager.connect();
+  }
+
+  @Test (groups = {"disabledOnTravis"})
+  //Test disabled on Travis because cluster integration tests are generally flaky on Travis.
+  public void testExecute() throws Exception {
+    suite.startCluster();
+
+    //Ensure that Helix has created a workflow
+    AssertWithBackoff.create().maxSleepMs(1000).backoffFactor(1).
+        assertTrue(ClusterIntegrationTest.isTaskStarted(helixManager, IntegrationJobSuite.JOB_ID), "Waiting for the job to start...");
+
+    //Instantiate config for HelixAssignedParticipantCheck
+    String helixJobId = Joiner.on("_").join(IntegrationJobSuite.JOB_ID, IntegrationJobSuite.JOB_ID);
+    helixConfig = helixConfig.withValue(GobblinClusterConfigurationKeys.HELIX_INSTANCE_NAME_KEY,
+        ConfigValueFactory.fromAnyRef(IntegrationBasicSuite.WORKER_INSTANCE_0))
+        .withValue(GobblinClusterConfigurationKeys.HELIX_JOB_ID_KEY, ConfigValueFactory.fromAnyRef(helixJobId))
+        .withValue(GobblinClusterConfigurationKeys.HELIX_PARTITION_ID_KEY, ConfigValueFactory.fromAnyRef(0));
+    HelixAssignedParticipantCheck check = new HelixAssignedParticipantCheck(helixConfig);
+
+    //Ensure that the SleepingTask is running
+    AssertWithBackoff.create().maxSleepMs(100).timeoutMs(2000).backoffFactor(1).
+        assertTrue(ClusterIntegrationTest.isTaskRunning(IntegrationJobSuite.TASK_STATE_FILE),"Waiting for the task to enter running state");
+
+    //Run the check. Ensure that the configured Helix instance is indeed the assigned participant
+    // (i.e. no exceptions thrown).
+    check.execute();
+
+    //Create Helix config with invalid partition num. Ensure HelixAssignedParticipantCheck fails.
+    helixConfig = helixConfig.withValue(GobblinClusterConfigurationKeys.HELIX_PARTITION_ID_KEY, ConfigValueFactory.fromAnyRef(1));
+    check = new HelixAssignedParticipantCheck(helixConfig);
+
+    try {
+      check.execute();
+      Assert.fail("Expected to throw CommitStepException");
+    } catch (CommitStepException e) {
+      //Do nothing; Expected to throw exception
```

Comment:
Shall we do an assertion here to make sure the exception being caught is indeed `CommitStepException `?

Reply:
Addressed.